packages/nextjs/src/config/loaders/valueInjectionLoader.ts

```diff
@@ -0,0 +1,26 @@
+import { LoaderThis } from './types';
+
+type LoaderOptions = {
+  values: Record<string, unknown>;
+};
+
+/**
+ * Set values on the global/window object at the start of a module.
+ *
+ * Options:
+ *   - `values`: A record where the keys correspond to the keys of the global values to set and the values
+ *        correspond to the values of the values on the global object. Values must be JSON serializable.
+ */
+export default function valueInjectionLoader(this: LoaderThis<LoaderOptions>, userCode: string): string {
+  // We know one or the other will be defined, depending on the version of webpack being used
+  const { values } = 'getOptions' in this ? this.getOptions() : this.query;
+
+  // Define some global proxy that works on server and on the browser.
+  let injectedCode = 'var _sentryCollisionFreeGlobalObject = typeof window === "undefined" ? global : window;\n';
```

Comment:
M: Do we need to worry about web workers/`self` here, do you think? Is there a reason not to just use `GLOBAL_OBJ` from `@sentry/utils`?

Reply:
I *think* we don't need to worry about web workers here. The loader is only applied to `sentry.server.config.js` and `sentry.client.config.js` and I *think* we are safe to assume that `sentry.client.config.js` will not be injected into any web worker contexts. Unless I am missing something.

Considering the above I would just keep it as-is to keep the logic simple.